### DISCUSSION TITLE
Any way to disable "Win" key input in a fullscreen window?

### DISCUSSION THREAD

**Discussion by cpfy** [AvaloniaUI/Avalonia#16895] [Q&A] [answered]
Hello!

**Thank you** for creating and maintaining Avalonia! It's a fantastic framework and I'm really enjoying working with it.

**I'm new to avalonia** and I'm trying to disable the `Win` key input within a fullscreen window in my application. Specifically, I'm testing on Windows and want to prevent the `Win` key from triggering system functions (like opening the Start Menu) that would overlay my fullscreen window.

**I've tried using the `OnKeyDown` method** to capture keyboard input, but it seems like the `Win` key has a higher system priority than Avalonia, and my code isn't able to intercept it.

**Is there a workaround for this issue?** For example:
- Could I bind the "Win" key to an empty hotkey within Avalonia?
- Is there a way to intercept the "Win" key input before it's handled by the system and prevent it from being sent?

Any guidance or examples on how to achieve this would be greatly appreciated.

Thanks again for your time and effort in developing Avalonia!

**Additional context:**
- Avalonia version: 11.1.2
- Target platform: Windows

**Comment by workgroupengineering** [comment]
It is not related to Avalonia. Win key is handle from Window OS  before  send to avalonia app.
Possible soluino are

- Replace Shell with your app https://superuser.com/a/1016381
- Windows Kios-Mode

**Reply by cpfy** [reply]
Thanks for your information! I'll give it a try.

**Comment by StefanKoell** [comment, marked as answer]
I can't speak for keyboard handling on the Mac or on Linux but in Windows you need Windows native API (low level keyboard hook) to handle this kind of things. Here's an older article which shows how to setup these hooks but I think it's still valid:
https://www.codeproject.com/articles/14485/low-level-windows-api-hooks-from-c-to-stop-unwante

**Reply by cpfy** [reply]
Thanks a lot! This actually solved my problem!

I rearrange the code, and hope this code snippet can help others who meet the same problem: https://gist.github.com/cpfy/a090d4542a1dabae55cc9a029c160aeb

**Comment by tobyfirth** [comment]
Something like Microsoft PowerToys Keyboard Manager or AutoHotkey will probably allow you to override the `Win` key.
